bindings/go/input/dir/method.go

```diff
@@ -106,3 +116,31 @@ func (i *InputMethod) ProcessSource(ctx context.Context, src *constructorruntime
 		ProcessedBlobData: fileBlob,
 	}, nil
 }
+
+// ensureWdOrPath ensures that the dirPath is absolute.
+// If the path is relative, it prepends the working directory to it.
+// If the working directory is empty, it does nothing.
+func ensureWdOrPath(dirPath, workingDirectory string) string {
```

Comment:
Im wondering about this. Can there be a case in which the dirPath is relative but outside of the workingDirectory that is set?

Example:

workingDirectory: "my/working/dir"
dirPath: "../../../path"

how would you resolve this now? shouldn't you create a root from "my/working/dir" and then ensure that you can open dirPath from within that root? That way you can avoid path escalation

Reply:
> Im wondering about this. Can there be a case in which the dirPath is relative but outside of the workingDirectory that is set?

Is this a problem? Why would we want to sandbox the users directories?

Reply:
because this can lead to attack vectors if not explicitly enabled. IMO we should not default to a potentially insecure "escape-allow". If we do allow this imo it should be opt-in, not opt-out